samples/SimpleDataSource/HelloWorldCustomDataProcessor.cs

```diff
@@ -0,0 +1,159 @@
+           ProcessorOptions options,
+           IApplicationEnvironment applicationEnvironment,
+           IProcessorEnvironment processorEnvironment,
+           IReadOnlyDictionary<TableDescriptor, Action<ITableBuilder, IDataExtensionRetrieval>> allTablesMapping,
+           IEnumerable<TableDescriptor> metadataTables)
+            : base(options, applicationEnvironment, processorEnvironment, allTablesMapping, metadataTables)
+        {
+            //
+            // Assign the files array to a readonly backing field.
+            //
+
+            this.filePaths = filePaths;
+        }
+
+        public override DataSourceInfo GetDataSourceInfo()
+        {
+            //
+            // The DataSourceInfo is used to tell the SDK the time range of the data(if applicable) and any other relevant data for rendering / synchronizing.
+            // In our case, we have nothing, so just return new DataSourceInfo(0, long.MaxValue) to tell the SDK that it encompasses all time.
+            //
+
+            return this.dataSourceInfo;
+            
+        }
+
+        protected override Task ProcessAsyncCore(
+           IProgress<int> progress,
+           CancellationToken cancellationToken)
+        {
+            //
+            // This is where you add your own logic to process the data into a format for your tables.
+            //
+            // In this sample, our tables will operate on a collection of lines in the file
+            // so we read all of the lines of the file and store them into a backing dictionary field.
+            //
+            // ProcessAsync is also where you would determine the information necessary for GetDataSourceInfo().
+            // In this sample, we take down the start and stop timestamps from the files
+            //
+            // Note: if you must do processing based on which tables are enabled, you would check the EnabledTables property (provided in the base class) on your class to see what you should do.
+            // For example, XPerf looks at what tables are enabled in order to turn on additional event processors to avoid processing everything if it doesn't have to.
+            //
+            // In this sample we create a Time Stamp column that can be used for graphing. For more information on graphing, go to "Graphing" on our Wiki. Link can be found in README.md
+            //
+
+            Timestamp startTime = Timestamp.MaxValue;
+            Timestamp endTime = Timestamp.MinValue;
+            DateTime firstEvent = DateTime.MinValue;
+            var contentDictionary = new Dictionary<string, IReadOnlyList<Tuple<Timestamp, string>>>();
+
+            foreach (var path in this.filePaths)
+            {
+                var list = new List<Tuple<Timestamp, string>>();
+                var content = System.IO.File.ReadAllLines(path);
+                
+
+                foreach (var line in content)
+                {
+                    var items = line.Split(new[] { ',' }, 2);
+
```

Comment:
Explain how exceptions are handled by the SDK; i.e. what happens if this method throws an exception. Do we crash? Is the processor/tabled disabled? Show blank data?

Reply:
The caller can choose what to do with the exception, it will be bubbled up by the SDK runtime

We should probably talk about this when using the runtime section